R/get-datasets.R

```diff
@@ -44,18 +48,28 @@ datasets <- function (x=getAPIRoot()) {
 #' the primary dataset catalog for the user will be used.
 #' @param refresh logical: should the function check the Crunch API for new
 #' datasets? Default is FALSE.
+#' @param shiny logical: launch a shiny gadget to help select the right dataset. The
+#' gadget will return a valid `loadDataset()` call which loads the selected dataset.
 #' @return Character vector of dataset names, each of which would be a valid
 #' input for [loadDataset()]
 #' @export
 listDatasets <- function (kind=c("active", "all", "archived"), project=NULL,
-                          refresh=FALSE) {
-    dscat <- selectDatasetCatalog(kind, project, refresh)
-    return(names(dscat))
+    refresh=FALSE, shiny = FALSE) {
+    if (shiny) {
+        checkInstalledPackages(c("rstudioapi", "shiny", "miniUI"))
+        if (rstudioapi::isAvailable("0.99.878")) {
```

Comment:
If you use `rstudioapi::verifyAvailable`, you don't need this `if`: https://github.com/rstudio/rstudioapi/blob/55d3200e6c8b1627b204215341501ca6e4e03212/R/code.R#L23